aea/crypto/fetchai.py

```diff
@@ -202,3 +202,36 @@ def is_transaction_settled(self, tx_digest: str) -> bool:
             # TODO: check the amount of the transaction is correct
             is_successful = True
         return is_successful
+
+    def validate_transaction(
+        self,
+        tx_digest: str,
+        seller: Address,
+        client: Address,
+        tx_nonce: str,
+        amount: int,
+    ) -> bool:
+        """
+        Check whether a transaction is valid or not.
+
+        :param seller: the address of the seller.
+        :param client: the address of the client.
+        :param tx_nonce: the transaction nonce.
+        :param amount: the amount we expect to get from the transaction.
+        :param tx_digest: the transaction digest.
+
+        :return: True if the random_message is equals to tx['input']
+        """
+
```

Comment:
the ledger api allows you to recover a list of transactions as well. It would be good if you could have a look at implementing it for fetch.ai too

Reply:
use: https://github.com/fetchai/ledger-api-py/blob/master/fetchai/ledger/api/tx.py#L113

Reply:
I think that we should add fetchai part in a different PR